File: cse241_OOP/hw7/driver2.java
import java.util.Random;
import java.util.Scanner;

public class driver2{

	public static void main(String[] args){
			
		try{
				
			int row = 0, column = 0,sayac=0;
			int tetro_n=0, kaydirma=0, kay=0, donus_s=0;
			char pre_shape , donus;
			Random rand = new Random();
			int aralik;
			Scanner inpt = new Scanner(System.in);

			System.out.printf("\n\n\n Give the size of Tetris board (row )\n"); //cout << endl << endl << endl << " Give the size of Tetris board (row )"<< endl;

			row = inpt.nextInt();

			column = row;
			Tetris board1 = new Tetris(row,column);
			board1.print_board();

			do{
				
				System.out.printf(" give the tetromino type\n");//cout << "give the tetromino type" << endl;

				do{
					pre_shape = inpt.next().charAt(0) ;
					
					if(pre_shape == 'R'){
						String alphabet = "TSLJZIO";
						pre_shape = (alphabet.charAt(rand.nextInt(alphabet.length())));
					}
					
					if(!board1.isLegal(pre_shape)) System.out.printf(" give the legal tetromino type \n"); //cerr << "Give the legal tetromno type" << endl;
				
				} while(!board1.isLegal(pre_shape));
				
				if(pre_shape == 'Q') break;	
				
				Tetromino obje = new Tetromino(pre_shape);
				
				board1.add_tetro_tomap(obje); // += obje;
				board1.print_board();

				donus_s = rand.nextInt()%4;
				String alphabet = "lr"	;
				donus = (alphabet.charAt(rand.nextInt(alphabet.length())));
				
				if(donus == 'l')
					for(int a = 0; a<donus_s; a++)
						obje.rotate_l();

				if(donus == 'r') 
					for(int a = 0; a<donus_s; a++)
						obje.rotate_r();
					
					
				board1.add_tetro_tomap(obje);//+= obje;
				board1.print_board();
				donus_s = rand.nextInt(10) % (row/2-1);
				
				switch(donus){
					case 'l':
						for(int a=0; a<donus_s; a++) board1.kaydir_sola(obje);
					break;
					case 'r':
						for(int a=0; a<donus_s; a++) board1.kaydir_saga(obje);
					break;
				
				}
				board1.animate(obje);
				if(board1.sil()) board1.print_board();	
			}while(!board1.is_finished());
			Thread.sleep(500);
		}
		catch(InterruptedException e){
		
		
		
		}
	}

}

File: cse241_OOP/hw3/driver2.cpp
#include<iostream>
#include "Tetromino.h"
#include "Tetris.h"
using namespace std;
using namespace Tetromino1;
using namespace Tetris1;


int main()
{

srand(time(NULL));

sleep_for(std::chrono::milliseconds(700));


int row = 0, column = 0,sayac=0;
int tetro_n=0, kaydirma=0, kay=0, donus_s=0;
char pre_shape , donus;

cout << endl << endl << endl << " size of Tetris board is 20X20"<< endl;
row = 20;

column = row;
Tetris board1(row,column);
board1.print_board();


	cout << "First tetromino is L" << endl;
	pre_shape = 'L';
	
	cout << " It is added with operator +=..." << endl;
	
	sleep_for(std::chrono::milliseconds(400));
	
	Tetromino obje(static_cast <tetro_shapes> (pre_shape));
	board1 += obje;
	board1.print_board();

	donus = 'l';
	donus_s = 2;

	if(donus == 'l')
		for(int a = 0; a<donus_s; a++)
			obje.rotate_l();

	if(donus == 'r') 
		for(int a = 0; a<donus_s; a++)
			obje.rotate_r();
	

	board1 += obje;
	board1.print_board();
	
	cout << "The turn direction is left and it turned two times... "<< endl;

	sleep_for(std::chrono::milliseconds(500));

	donus = 'l' ;
	donus_s = 9;
	cout << "it will go to left side 9 times" << endl;
	sleep_for(std::chrono::milliseconds(500));


	switch(donus){
	
		case 'l':
		
		for(int a=0; a<donus_s; a++) board1.kaydir_sola(obje);
		
		
		break;
	
	
		case 'r':
		
		for(int a=0; a<donus_s; a++) board1.kaydir_saga(obje);
		
		break;
	
	}
	
	sleep_for(std::chrono::milliseconds(500));

	board1.animate(obje);
	
	if(board1.sil()) board1.print_board();	
	
	cout << "Additional test.." << endl;
	pre_shape = 'T';
	
	Tetromino obje1(static_cast <tetro_shapes> (pre_shape));
	board1 += obje1;
	board1.print_board();

	donus = 'l';
	donus_s = 2;

	if(donus == 'l')
		for(int a = 0; a<donus_s; a++)
			obje1.rotate_l();

	if(donus == 'r') 
		for(int a = 0; a<donus_s; a++)
			obje1.rotate_r();
	

	board1 += obje1;
	board1.print_board();

	sleep_for(std::chrono::milliseconds(250));

	donus = 'r' ;
	donus_s = 7;
	sleep_for(std::chrono::milliseconds(250));


	switch(donus){
	
		case 'l':
		
		for(int a=0; a<donus_s; a++) board1.kaydir_sola(obje1);
		
		
		break;
	
	
		case 'r':
		
		for(int a=0; a<donus_s; a++) board1.kaydir_saga(obje1);
		
		break;
	
	}
	
	sleep_for(std::chrono::milliseconds(250));

	board1.animate(obje1);
	

	

return 0;
}
